new resource:`azurerm_application_load_balancer_association` (#23628)
hashicorp/terraform-provider-azurerm: website/docs/r/application_load_balancer_association.html.markdown

> ---

Reviewer: and this file name?

subcategory: "Service Networking"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_application_load_balancer_association"
description: |-
  Manages an Application Gateway for Containers Association.
---

# azurerm_application_load_balancer_association

> Manages an Application Gateway for Containers Association.

Reviewer 2: since this is a _Subnet_ association, this resource probably should be `_subnet_association`?

## Example Usage

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-rg"

>   location = "northeurope"

Reviewer 3: ```suggestion
  location = "westeurope"
```

}

resource "azurerm_application_load_balancer" "example" {
  name                = "example-alb"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_virtual_network" "example" {
  name                = "example-vnet"
  address_space       = ["10.0.0.0/16"]
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_subnet" "example" {
  name                 = "example-subnet"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefixes     = ["10.0.1.0/24"]

  delegation {
    name = "delegation"

    service_delegation {
      name    = "Microsoft.ServiceNetworking/trafficControllers"
      actions = ["Microsoft.Network/virtualNetworks/subnets/join/action"]
    }
  }
}

resource "azurerm_application_load_balancer_subnet_association" "example" {
  name                         = "example"
  application_load_balancer_id = azurerm_application_load_balancer.example.id

>   subnet_id                    = azurerm_subnet_example.id

Reviewer 4: ```suggestion
  subnet_id                    = azurerm_subnet.example.id
```

}
```

## Arguments Reference

The following arguments are supported:

* `name` - (Required) The name which should be used for this Application Gateway for Containers Association. Changing this forces a new resource to be created.

* `application_load_balancer_id` - (Required) The ID of the Application Gateway for Containers. Changing this forces a new resource to be created.

* `subnet_id` - (Required) The ID of the subnet which the Application Gateway for Containers associated to. Changing this forces a new resource to be created.

> **Note:** The subnet should be delegated by `Microsoft.ServiceNetworking/trafficControllers` as the example. 

Reviewer 3: ```suggestion
**Note:** The subnet to be used must be have a delegation for  `Microsoft.ServiceNetworking/trafficControllers` as shown in the example above.
```
